File: app/src/main/java/com/onemoresecret/crypto/AesKeyAlgorithm.java
package com.onemoresecret.crypto;

/**
 * See https://docs.oracle.com/javase/7/docs/api/javax/crypto/Cipher.html for
 * the list of supported transformations
 *
 * See https://developer.android.com/reference/javax/crypto/Cipher for Android
 *
 */
public enum AesKeyAlgorithm {
    PBKDF2WithHmacSHA256("PBKDF2WithHmacSHA256");

    public final String keyAlgorithm;

    AesKeyAlgorithm(String transformation) {
        this.keyAlgorithm = transformation;
    }
}

File: app/src/main/java/com/onemoresecret/crypto/AesKeyAlgorithm.kt
package com.onemoresecret.crypto

/**
 * See [javax.crypto.Cipher Java API](https://docs.oracle.com/javase/7/docs/api/javax/crypto/Cipher.html) for
 * the list of supported transformations
 *
 * See [java.crypto.Cipher Android API](https://developer.android.com/reference/javax/crypto/Cipher)
 *
 */
enum class AesKeyAlgorithm(@JvmField val keyAlgorithm: String) {
    PBKDF2WithHmacSHA256("PBKDF2WithHmacSHA256");

}
